cmd/fossa/cmd/analyze/analyze.go

```diff
@@ -363,8 +375,53 @@ func uploadAnalysis(normalized []fossa.SourceUnit) error {
 	display.ClearProgress()
 	if err != nil {
 		log.Fatalf("Error during upload: %s", err.Error())
-		return err
+		return fossa.Locator{}, err
 	}
 	fmt.Println(locator.ReportURL())
-	return nil
+	return locator, nil
+}
+
+func fetchGitContibutors() map[string]string {
+	year, month, day := time.Now().UTC().AddDate(0, 0, -90).Date()
+	fmtSince := fmt.Sprintf("%d-%d-%d", year, month, day)
+
+	// the format arg produces newline-separated lines of: <author-email> :: <commit date>
+	// We use commit date since some people backdate authorship
+	// dates are forced into YYYY-MM-DD format.
+	cmd := exec.Cmd{
+		Name:    "git",
+		Argv:    []string{"log", "--since", fmtSince, "--format=%ae :: %cd", "--date=short"},
+		Timeout: "10s",
+	}
+	output, _, err := exec.Run(cmd)
+	if err != nil {
+		log.Debugf("Failed to run 'git log': %s", err.Error())
+		return nil
+	}
+
+	contribList := string(output)
+
+	contributorDates := make(map[string]string)
+	for _, line := range strings.Split(contribList, "\n") {
```

Comment:
Splitting on `\n` leaves any `\r` in the output (if present). It's recommended to instead use `bufio.Scanner`, which scans line-by-line by default. Example:
```go
scanner := bufio.NewScanner(bytes.NewReader(output))
for scanner.Scan() {
  line := scanner.Text()
  // ...
}
```